### DISCUSSION TITLE
Not able to make any smtp configuration work with self hosted fief

### DISCUSSION THREAD

**Discussion by maverick891** [fief-dev/fief#234] [Q&A] [answered]
I have a self hosted fief server in docker. Not able to get any smtp configuration to work. I tried postmark, sendgrid and my own smtp server. Nothing worked. Has anybody has any success?

**Comment by fief-bailiff** [comment]
Hail, @maverick891 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
Can you share the config you're trying? For SMTP, a possible configuration could look like this in your environment variables:

```ini
EMAIL_PROVIDER=SMTP
EMAIL_PROVIDER_PARAMS={"host": "smtp.bretagne.duchy", "username": "anne", "password": "herminetincture"}
```

**Reply by maverick891** [reply]
I finally got postmark to partially work. The problem is fief server is sending emails from fief.dev but I would like to send emails to my user from admin@mydomain.com. There is no setting to change this.

**Reply by frankie567** [reply]
Actually, there are settings for this (undocumented, sorry, that's a miss). So you can set those env vars:

```ini
DEFAULT_FROM_EMAIL=admin@mydomain.com
DEFAULT_FROM_NAME=My Company
```

EDIT: now they are -> https://docs.fief.dev/self-hosting/environment-variables/#email-provider

**Reply by maverick891** [reply]
Thank you for that and the prompt reply. I tried another SMTP configuration. I am able to send emails using this config with a python script but fief server was unable to use them to send a mail.

EMAIL_PROVIDER=SMTP
EMAIL_PROVIDER_PARAMS={"host": "smtppro.zoho.com", "username": "admin@afterword.tech", "password": "HIDDEN", "port": "465"}
DEFAULT_FROM_EMAIL=admin@afterword.tech
DEFAULT_FROM_NAME=Afterword

**Reply by frankie567** [reply, marked as answer]
Our SMTP implementation only supports TLS (instead of SSL —SMTP is complicated —). Can you try with port `587`? From what I see, Zoho should support it as well.

**Reply by maverick891** [reply]
Thanks a lot. That worked :)

**Reply by kartaris** [reply]
> Our SMTP implementation only supports TLS (instead of SSL —SMTP is complicated —). Can you try with port `587`? From what I see, Zoho should support it as well.

In my case there was no choice to change the destination port to 587
Think this could solve the discussed problem in future: #537
